### PR TITLE
update sample data and templates

### LINKED ISSUE #349: Update sample data MEF files and templates
I have been updating documentation and have noted some issues with our sample data:

1) The [update instructions](https://metadata101.github.io/iso19139.ca.HNAP/admin/upgrade.html) provide instructions to edit, save and close to resolve codeList references to ``http://nap.geogratis.gc.ca/metadata/register/napMetadataRegister.xml``.

2) The web services associated with the sample data are not working or are inconsistent

**Harbour Porpoise Presence, Maritimes Region**  `58ea48ab-f052-48ab-9c18-4353e51b8bea`

image
image

Thumbnail is included but not displayed

image

The **Add to map** links do not work
- The service appears to be available but not displayed in the map.
- The Add to map ends up zooming to NullIsland in the Atlantic, no visual is ever returned.
- navigating to the correct area does not display contents
- Javascript console says:
  
  > The layer has not been found in any group: Harbour Porpoise Presence MAR ARP Pilots
  > gn_search_default.js:627:132

- HNAP standard indicates that ttps://gisp.dfo-mpo.gc.ca/arcgis/rest/services/FGP/OPP_ARP_MAR_SpeciesPresence_HarbourPorpoise/MapServer/0 should be used; but that also does not work

image

The record formats "OGC:WMS" and "OGC:WFS" are supported
- Creating appropriate URL https://gisp.dfo-mpo.gc.ca/arcgis/rest/services/FGP/OPP_ARP_MAR_SpeciesPresence_HarbourPorpoise/MapServer/WMSServer?REQUEST=GetCapabilities is not configured as a WMS.

### PR DESCRIPTION
**Pull request by jodygarnett**
Resolve https://github.com/metadata101/iso19139.ca.HNAP/issues/349

* [x] Save and Load each record and template to apply schema and codeList changes
* [x] Correct **Harbour Porpoise Presence** thumbnail and add to map links
* [x] Two records that show how WMS links work
* [x] Sample data uses directory for ease of maintenance, pom.xml build to zip into *mif* files for plugin

After these changes a catalog loaded with HNAP sample data appears as follows:

image

### CODE REVIEW

File: src/main/plugin/iso19139.ca.HNAP/sample-data/Harbour Porpoise Presence/58ea48ab-f052-48ab-9c18-4353e51b8bea/metadata/metadata.xml
Comment (jodygarnett) [inline src/main/plugin/iso19139.ca.HNAP/sample-data/Harbour Porpoise Presence/58ea48ab-f052-48ab-9c18-4353e51b8bea/metadata/metadata.xml:894]: reference to single fra layer

File: src/main/plugin/iso19139.ca.HNAP/sample-data/Harbour Porpoise Presence/58ea48ab-f052-48ab-9c18-4353e51b8bea/metadata/metadata.xml
Comment (jodygarnett) [inline src/main/plugin/iso19139.ca.HNAP/sample-data/Harbour Porpoise Presence/58ea48ab-f052-48ab-9c18-4353e51b8bea/metadata/metadata.xml:864]: reference to single eng layer

File: pom.xml
Comment (jodygarnett) [inline pom.xml:131]: This is responsible for creating mef files

**Comment by jodygarnett** [comment]
Thanks for your help troubleshooting @josegar74 enjoy your vacation.

@ianwallen if you wish to test this out it is ready for review; there is no rush I just needed something that worked to write documentation.

File: src/main/plugin/iso19139.ca.HNAP/templates/hnap_nonspatial-english.xml
Comment (ianwallen) [inline src/main/plugin/iso19139.ca.HNAP/templates/hnap_nonspatial-english.xml]: Why are you adding this?  I thought this was causing issues because it is the main language?  We used to have it in the past and I thought we removed it because it was causing issues - I don't recall the issue exactly - Maybe @josegar74 remembers?

File: src/main/plugin/iso19139.ca.HNAP/templates/hnap_nonspatial-english.xml
Comment (ianwallen) [inline src/main/plugin/iso19139.ca.HNAP/templates/hnap_nonspatial-english.xml]: Why are you removing the locale code?  This line is not required if locale code is removed

File: src/main/plugin/iso19139.ca.HNAP/templates/hnap_nonspatial-english.xml
Comment (ianwallen) [inline src/main/plugin/iso19139.ca.HNAP/templates/hnap_nonspatial-english.xml]: This is incorrect.
We have the following PR to fix the issue but still waiting on Approval from Geocat?
https://github.com/metadata101/iso19139.ca.HNAP/pull/345

If you make this change, it will only cause conflicts
Is it possible to approve and merge that PR first and then update this file?

File: pom.xml
Comment (ianwallen) [reply pom.xml:131]: This is great :) - I always found it to be a pain to be committing zip files - nice to see that we can now version control individual files without having to commit a zip file.

File: src/main/plugin/iso19139.ca.HNAP/templates/hnap_nonspatial-english.xml
Comment (jodygarnett) [reply src/main/plugin/iso19139.ca.HNAP/templates/hnap_nonspatial-english.xml]: I followed the update instructions to load and safe things; there is a chance this process has introduced things we do not want included in a template.

But if someone uses this template the above will be added as they edit...

File: src/main/plugin/iso19139.ca.HNAP/templates/hnap_nonspatial-english.xml
Comment (jodygarnett) [reply src/main/plugin/iso19139.ca.HNAP/templates/hnap_nonspatial-english.xml]: Yes of course.

Done

File: pom.xml
Comment (jodygarnett) [reply pom.xml:131]: I should probably look at add-schema.sh a bit as it will only work when using the default unzipping the maven plugin artifact approach.

Still if you like this we could do it for other schema plugins also on core-geonetwork.

File: src/main/plugin/iso19139.ca.HNAP/templates/hnap_nonspatial-english.xml
Comment (jodygarnett) [reply src/main/plugin/iso19139.ca.HNAP/templates/hnap_nonspatial-english.xml]: I just did a load and save, perhaps that process is removing the code?

File: src/main/plugin/iso19139.ca.HNAP/templates/hnap_nonspatial-english.xml
Comment (jodygarnett) [inline src/main/plugin/iso19139.ca.HNAP/templates/hnap_nonspatial-english.xml:177]:
```suggestion
```

**Comment by jodygarnett** [comment]
I believe I have addressed all feedback, returning to documentation until I hear back.

File: src/main/plugin/iso19139.ca.HNAP/templates/hnap_nonspatial-french.xml
Comment (ianwallen) [inline src/main/plugin/iso19139.ca.HNAP/templates/hnap_nonspatial-french.xml]: Another case where the locale id was lost

File: src/main/plugin/iso19139.ca.HNAP/templates/hnaplevel1-french.xml
Comment (ianwallen) [inline src/main/plugin/iso19139.ca.HNAP/templates/hnaplevel1-french.xml]: Another case where the locale id was lost

File: src/main/plugin/iso19139.ca.HNAP/sample-data/Dynamic Radar Coverage/6b02c778-8eaa-46f5-8786-ae80b0ea0f72/metadata/metadata-iso19139.xml
Comment (ianwallen) [inline src/main/plugin/iso19139.ca.HNAP/sample-data/Dynamic Radar Coverage/6b02c778-8eaa-46f5-8786-ae80b0ea0f72/metadata/metadata-iso19139.xml]: There are lots (20+) of cases where the "theme" for the codeListValue is incorrect. - Now that you have merged the PR that fixes this issue, you many need to re-update these files? Or change them manually?

File: src/main/plugin/iso19139.ca.HNAP/templates/hnaplevel1-french.xml
Comment (jodygarnett) [reply src/main/plugin/iso19139.ca.HNAP/templates/hnaplevel1-french.xml]: I do not know what to make of that; if the process is losing these values then it will happen the first time someone makes content with the template, edits, and then saves?

I will fix these but expect we may have another issue to deal with...

File: src/main/plugin/iso19139.ca.HNAP/sample-data/Dynamic Radar Coverage/6b02c778-8eaa-46f5-8786-ae80b0ea0f72/metadata/metadata-iso19139.xml
Comment (jodygarnett) [reply src/main/plugin/iso19139.ca.HNAP/sample-data/Dynamic Radar Coverage/6b02c778-8eaa-46f5-8786-ae80b0ea0f72/metadata/metadata-iso19139.xml]: I will try another round of load/save now that the fix is available and report back...

Comment (jodygarnett) [reply src/main/plugin/iso19139.ca.HNAP/sample-data/Dynamic Radar Coverage/6b02c778-8eaa-46f5-8786-ae80b0ea0f72/metadata/metadata-iso19139.xml]: That appears to have addressed the issue

File: src/main/plugin/iso19139.ca.HNAP/templates/hnap_nonspatial-english.xml
Comment (jodygarnett) [reply src/main/plugin/iso19139.ca.HNAP/templates/hnap_nonspatial-english.xml]: This was the only document that did not already have both ...

Still if we wish to remove the gmd:locale matching the main language I will do so as a distinct commit: c4a2ab6ac829f1efb92b1a99ff84b20818b0ceb9

**Comment by jodygarnett** [comment]
rebased and cleaned up commit history; ready to go

**Review by ianwallen — APPROVED** [review]

**Pull request merged**
